### PR TITLE
Fix: passing wrong strings for scheduler interval doesn't throw an error

### LINKED ISSUE #5548: Passing wrong strings for scheduler interval doesn't throw an error
I noticed that when you pass an interval as scheduler parameter that is not 'step' or 'epoch' (for example 'batch') there's no error, it just trains without ever calling the scheduler. Had to look into the source code because I didn't find much info about it in the documentation

### PR DESCRIPTION
**Pull request by ddrevicky**
## What does this PR do?
- Raises `MisconfigurationException` if scheduler's interval parameter is not one of `epoch`, `step`
- Adds a test case to verify this behavior

Fixes #5548 
## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/trainer/optimization/test_optimizers.py [hunk 1]
```diff
@@ -459,6 +459,24 @@ def test_unknown_configure_optimizers_raises(tmpdir):
         trainer.fit(model)
 
 
+def test_lr_scheduler_with_unknown_interval_raises(tmpdir):
+    """
+    Test exception when lr_scheduler dict has unknown interval param value
+    """
+    model = EvalModelTemplate()
```
Comment (Borda) [inline tests/trainer/optimization/test_optimizers.py:466]:
```suggestion
    model = BoringModel()
```

File: pytorch_lightning/core/datamodule.py [hunk 2]
```diff
@@ -15,10 +15,9 @@
 
 import functools
 import inspect
-import os
 from abc import abstractmethod
 from argparse import ArgumentParser, Namespace
-from typing import Any, List, Optional, Tuple, Union, Dict, Sequence, Mapping
+from typing import Any, List, Mapping, Optional, Sequence, Tuple, Union
```
Comment (ddrevicky) [inline pytorch_lightning/core/datamodule.py:20]: isort was failing in CI because of import ordering in `datamodule.py`, not sure why since I did not originally modify the file at all. But since I had to fix the import order there, I also removed unused imports and applied yapf while I was at it.

Comment (ddrevicky) [reply pytorch_lightning/core/datamodule.py:20]: I also had to fix `test_datamodules.py`, I have no idea why :) There is an unused class `CustomMNISTDataModule` there by the way.

File: pytorch_lightning/trainer/optimizers.py [hunk 3]
```diff
@@ -117,6 +117,13 @@ def configure_schedulers(self, schedulers: list, monitor: Optional[str] = None):
                     raise MisconfigurationException(
                         'The lr scheduler dict must have the key "scheduler" with its item being an lr scheduler'
                     )
+                allowed_interval_keys = ['step', 'epoch']
+                if 'interval' in scheduler and scheduler['interval'] not in allowed_interval_keys:
```
Comment (carmocca) [inline pytorch_lightning/trainer/optimizers.py:121]:
```suggestion
                if 'interval' in scheduler and scheduler['interval'] not in ('step', 'epoch'):
```

File: pytorch_lightning/core/datamodule.py (hunk 2 above)
Comment (carmocca) [reply pytorch_lightning/core/datamodule.py]: > There is an unused class CustomMNISTDataModule there by the way.

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/core/datamodule.py (hunk 2 above)
Comment (rohitgr7) [reply pytorch_lightning/core/datamodule.py]: already removed.

File: pytorch_lightning/trainer/optimizers.py (hunk 3 above)
Comment (rohitgr7) [reply pytorch_lightning/trainer/optimizers.py]: `allowed_interval_keys` is used in error message. need to revert the changes here I guess. @ddrevicky

**Review by SkafteNicki — APPROVED** [review]

File: pytorch_lightning/trainer/optimizers.py (hunk 3 above)
Comment (carmocca) [reply pytorch_lightning/trainer/optimizers.py]: Oops, my bad 🙈

**Review by rohitgr7 — APPROVED** [review]
